src/github.com/getlantern/detour/whitelist.go

```diff
@@ -15,47 +14,56 @@ var (
 	whitelist   = make(map[string]wlEntry)
 )
 
-func InitWhitelist(wl map[string]time.Time) {
+// AddToWl adds a domain to whitelist, all subdomains of this domain
+// are also considered to be in the whitelist.
+func AddToWl(addr string, permanent bool) {
 	muWhitelist.Lock()
 	defer muWhitelist.Unlock()
-	for k, v := range wl {
-		whitelist[k] = wlEntry{true, v}
-	}
-	return
+	whitelist[addr] = wlEntry{permanent}
 }
 
-func DumpWhitelist() (wl map[string]time.Time) {
-	wl = make(map[string]time.Time)
+func RemoveFromWl(addr string) {
+	muWhitelist.Lock()
+	defer muWhitelist.Unlock()
+	delete(whitelist, addr)
+}
+
+func DumpWhitelist() (wl []string) {
```

Comment:
`DumpWhitelist` is also unused. But it may has potential usage perhaps, so I just leave it here, but I changed the signature.